### PR TITLE
Don't try to stop Bugsnag if it's not needed

### PR DESCRIPTION
**Pull request by nimeacuerdo**
Use the same flag already in place for starting (or not) Bugsnag in order to decide if it needs to be stopped too.

### CODE REVIEW

File: cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt [hunk 1]
```diff
@@ -59,7 +59,9 @@ fun main(args: Array<String>): Unit = mainBody(
             }
         } finally {
             stopKoin()
-            bugsnagExceptionsReporter.end()
+            if(bugsnagReporting){
```
Comment (Malinskiy) [inline cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt]: We're using the IntelliJ's automatic code style formatting, would you be able to hit the auto-format in the IDE? Alternatively, a space after `if` and after the closing bracket `)`.
Thank you.

**Review by Malinskiy — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt (hunk 1 above)
Comment (nimeacuerdo) [reply cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt]: Sure, I just copied the existing approach from https://github.com/Malinskiy/marathon/blob/063a759ed9d416039936b95a71651fbe9abfa693/cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt#L36, which I guess should be updated too, right?

Comment (Malinskiy) [reply cli/src/main/kotlin/com/malinskiy/marathon/cli/ApplicationView.kt]: Yeah, seems fair. Thanks!

**Comment by Malinskiy** [comment]
The change also needs to be in the gradle plugin: https://github.com/Malinskiy/marathon/blob/develop/marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonRunTask.kt#L100

I'm leaning towards having this if condition inside the BugsnagExceptionsReporter. Maybe passing the `enabled` configuration option as a constructor parameter and then setting the `private val bugsnag: Bugsnag? by lazy` to be null if it's not enabled. What do you think?

**Comment by nimeacuerdo** [comment]
> The change also needs to be in the gradle plugin: https://github.com/Malinskiy/marathon/blob/develop/marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonRunTask.kt#L100
> 
> I'm leaning towards having this if condition inside the BugsnagExceptionsReporter. Maybe passing the `enabled` configuration option as a constructor parameter and then setting the `private val bugsnag: Bugsnag? by lazy` to be null if it's not enabled. What do you think?

sure, adding new changes.

**Review by Malinskiy — APPROVED** [review]

**Comment by Malinskiy** [comment]
@nimeacuerdo thank you for your contribution 👍

**Pull request merged**